src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -341,6 +366,12 @@ contract ECDSAStakeRegistry is
         if (newSigningKey == oldSigningKey) {
             return;
         }
+        // Remove the old signing key from the mapping if it exists
+        if (oldSigningKey != address(0)) {
+            _signingKeyToOperatorHistory[oldSigningKey].push(uint160(0));
+        }
+        // Update the new signing key to point to this operator
+        _signingKeyToOperatorHistory[newSigningKey].push(uint160(operator));
```

Comment:
This doesn't check for conflict like you do on registration.

What if I update to someone else's signing key (that is `_signingKeyToOperatorHistory[newSigningKey]` is not 0). That should error as well.

Please add a test and add the check